Remove library helpers.

damacus opened the pull request:
- Moves the escape_string into the resources using it
- Everything still left using escape_regexp now uses escape string

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sous-chefs/line-cookbook/78)
<!-- Reviewable:end -->

damacus committed:
Remove name_property for name, as this is actually an error

Commit:
Merge branch 'master' of https://github.com/sous-chefs/line-cookbook

Commit:
Remove reference to match_count from the file_ext matcher

Match_count was moved to a separate matcher

Commit:
Delete minitest files

Minitest is no longer used for integration tests. Delete the leftover files.

Commit:
Merge branch 'master' of https://github.com/sous-chefs/line-cookbook

Commit:
Merge branch 'master' of https://github.com/sous-chefs/line-cookbook

Commit:
Handle missing file errors better

The code raised an error for a missing target file, but did the raise inside of a recovery block.
The recovery block would open a temporary file and ensure that the temporary file would be closed and
unlinked.  Since the temporary file never was created the the close failed obscuring the cause of the
error.

Commit:
Add tests for resource calls with missing files

The delete_lines, add_to_list, and delete_from_list resources should
raise an error if the target file has not been created. The added
tests verify that the right error and message is created.

Commit:
Version 1.0.3

Additional tests.
Fix missing file bugs.

Commit:
Handle a delete item edge case

Spaces as delimiters cause problems when deleting itermediate items in a list.
Make sure a space gets left behind when an item in a space delimited list is
deleted.

Commit:
Version 1.0.4

Commit:
Merge branch 'Delete_item_space_delim'

damacus committed:
Remove library helpers.
- Moves the escape_string into the resources using it
- Everything still left using escape_regexp now uses escape string

damacus committed:
Add raise if file not found to every resource

damacus committed:
Replace or add actually throws a new file down on disk. This seems like
odd behaviour for replace_or_add. It doesn't mention anything about
creating a new file.

damacus committed:
Cookstyle

MarkGibbons commented on resources/replace_or_add.rb:
```diff
@@ -53,19 +53,7 @@
-      end
-    end
-  end
+
```
Creating the new file was a deliberate choice.  We are doing an add.

damacus replied:
Noted reverting that change 👍

damacus committed:
Update CHANGELOG

MarkGibbons commented on resources/replace_or_add.rb:
```diff
@@ -1,12 +1,12 @@
 
 action :edit do
-  regex = /#{new_resource.pattern}/
+  regex = new_resource.pattern.is_a?(String) ? /#{new_resource.pattern}/ : new_resource.pattern
```
nice

damacus replied:
I'm just standardising that pattern where I can. If/when we remove it, it'll be the same change rather than guessing who, what, why.

damacus committed:
Remvoe unused helper

MarkGibbons commented on resources/add_to_list.rb:
```diff
@@ -25,7 +25,7 @@
     regexdelim = []
     new_resource.delim.each do |delim|
-      regexdelim << escape_regex(delim)
+      regexdelim << escape_string(delim)
```
Would this work here: string = Regexp.escape(delim)
The we could get rid of the escape string methods.
Pretty sure this is what Lamont did in append if no line.

damacus committed:
Remove escape_string everywhere

MarkGibbons commented on resources/delete_from_list.rb:
```diff
@@ -1,12 +1,12 @@
 property :path, String
```
The tests are failing on delete_from_list.  Those are the most fragile and incomplete tests we have in this cookbook. I can look at the tests and failures tomorrow.  Have to deal with some other stuff the rest of today.

Commit:
Merge branch 'remove_library_helpers' of https://github.com/sous-chefs/line-cookbook into remove_library_helpers

MarkGibbons commented on resources/delete_from_list.rb:
```diff
@@ -35,7 +35,7 @@
         case line
         when /#{regexdelim[0]}\s*#{new_resource.entry}/
-          replace_deleted = regexdelim[0] =~ /\s+/ ? regexdelim[0] : ''
+          replace_deleted = regexdelim[0] =~ /\s+/ ? new_resource.delim[0] : ''
```
This change fixed the failures. But the check regexdelim[0] =~ /\s+/ probably fails in some edge case. I'll try to come up with some better tests.

Commit:
Clean up edge cases for delete from list

Don't leave trailing delimeters on a list
Remove trailing white space from lists

MarkGibbons reviewed (approved):
Looks good to me.

damacus commented on resources/delete_from_list.rb:
```diff
@@ -35,10 +35,10 @@
-          line = line.sub(/(#{regexdelim[0]})*\s*#{new_resource.entry}(#{regexdelim[0]})*/, replace_deleted)
-          line = line.chomp
-          line = line.rstrip if regexdelim[0] =~ /\s+/
+          # remove the entry
```
Nice. I'm finding it hard to reason what this resource is doing.
